lib/presentation/screens/home/home.dart

```diff
@@ -12,28 +14,76 @@ class MyHomePage extends StatefulWidget {
 }
 
 class _MyHomePageState extends State<MyHomePage> {
+  final postingsRepo =
+      PostingsRepositoryImpl(firestore: PostingFirestoreImpl());
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      body: ListView(
-        padding: const EdgeInsets.all(16.0),
-        children: [
-          Text(
-            InstrealLocalizations.of(context)!.whatsUpIn,
-            style: Theme.of(context).textTheme.displayLarge,
-          ),
-          Text(
-            InstrealLocalizations.of(context)!.montreal,
-            style: Theme.of(context).textTheme.displayLarge!.copyWith(
-                  color: Theme.of(context).colorScheme.primary,
-                  fontFamily: GoogleFonts.caveat().fontFamily,
+      body: NestedScrollView(
+        headerSliverBuilder: (context, innerBoxIsScrolled) => [
+          SliverAppBar(
+            expandedHeight: 160,
+            collapsedHeight: 160,
+            flexibleSpace: FlexibleSpaceBar(
+              background: Padding(
+                padding: const EdgeInsets.all(16.0),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Text(
+                      InstrealLocalizations.of(context)!.whatsUpIn,
+                      style: Theme.of(context).textTheme.displayLarge,
+                    ),
+                    Text(
+                      InstrealLocalizations.of(context)!.montreal,
+                      style: Theme.of(context).textTheme.displayLarge!.copyWith(
+                            color: Theme.of(context).colorScheme.primary,
+                            fontFamily: GoogleFonts.caveat().fontFamily,
+                          ),
+                    ),
+                  ],
                 ),
+              ),
+            ),
+            floating: false,
+            pinned: false,
           ),
-          const PostCard(),
-          const PostCard(),
-          const PostCard(),
-          const PostCard(),
         ],
+        body: FutureBuilder<List<Posting>>(
+          future: postingsRepo.postings,
+          builder: (context, snapshot) => switch (snapshot) {
+            AsyncSnapshot(connectionState: ConnectionState.none) ||
+            AsyncSnapshot(connectionState: ConnectionState.waiting) ||
+            AsyncSnapshot(connectionState: ConnectionState.active) =>
```

Comment:
Why don't you use else instead? :)

Reply:
haha. yeah. Initially, I wanted to put different behavior but ended up having only 2.
updated it.